workshops/views.py

```diff
@@ -208,9 +207,17 @@ def form_valid(self, form):
 #------------------------------------------------------------
 
 
-@login_required
+def dispatch(request):
+    if request.user and is_admin(request.user):
+        return redirect(reverse('dashboard'))
+    else:
+        return redirect(reverse('trainee-dashboard'))
```

Comment:
I generally dislike "significant omissions": if one is called `trainee-dashboard`, the other should be called `admin-dashbord`, not just `dashboard`.